### PR TITLE
Kernel abs

### PR DESCRIPTION
**Pull request by Rubtsowa**

### CODE REVIEW

File: dpnp/backend/custom_kernels_mathematical.cpp [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+template <typename _KernelNameSpecialization>
+class custom_elemwise_absolute_c_kernel;
+
+template <typename _DataType>
+void custom_elemwise_absolute_c(void* array1_in,
+                                 const std::vector<long>& input_shape,
+                                 void* result1,
+                                 size_t size)
+{
+    if (!size)
```
[the remark this answers is not in the document]
Comment (shssf) [reply dpnp/backend/custom_kernels_mathematical.cpp:44]: @samir-nasibli it should be here. Let's imagine if somebody will call this kernels from C++ application.

File: dpnp/backend/custom_kernels_mathematical.cpp [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+#include <backend_iface.hpp>
+#include "backend_utils.hpp"
+#include "queue_sycl.hpp"
+
+template <typename _KernelNameSpecialization>
+class custom_elemwise_absolute_c_kernel;
+
+template <typename _DataType>
+void custom_elemwise_absolute_c(void* array1_in,
+                                 const std::vector<long>& input_shape,
```
Comment (Alexander-Makaryev) [inline dpnp/backend/custom_kernels_mathematical.cpp:40]: @Rubtsowa Please avoid of using complex types like std::vector in interface, if you need to provide set of values then use pointers. At this moment it is hard to use such params in calls from another projects like Numba, for example. We already had discussions about it.
But in this function this param is useless at all. You don't need it and you didn't use this param, look at another comments.

File: dpnp/backend/custom_kernels_mathematical.cpp [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        return;
+    }
+
+    cl::sycl::event event;
+    _DataType* array1 = reinterpret_cast<_DataType*>(array1_in);
+    _DataType* result = reinterpret_cast<_DataType*>(result1);
+
+    const size_t input_shape_size = input_shape.size();
+    size_t* input_offset_shape = reinterpret_cast<size_t*>(dpnp_memory_alloc_c(input_shape_size * sizeof(long)));
+    size_t* result_offset_shape = reinterpret_cast<size_t*>(dpnp_memory_alloc_c(input_shape_size * sizeof(long)));
```
Comment (Alexander-Makaryev) [inline dpnp/backend/custom_kernels_mathematical.cpp:55]: @Rubtsowa This code is useless. BTW it is only place where input_shape is used.

File: dpnp/backend/backend_iface_fptr.hpp [hunk 4]
```diff
@@ -59,7 +59,8 @@
  */
 enum class DPNPFuncName : size_t
 {
-    DPNP_FN_NONE,      /**< Very first element of the enumeration */
+    DPNP_FN_NONE,     /**< Very first element of the enumeration */
+    DPNP_FN_ABSOLUTE,      /**< Used in numpy.absolute() implementation  */
```
Comment (Alexander-Makaryev) [inline dpnp/backend/backend_iface_fptr.hpp:63]: @Rubtsowa These rows show that codestyle was not applied. Please use codestyle tools.

File: dpnp/backend/backend_iface.hpp [hunk 5]
```diff
@@ -97,6 +97,26 @@ template <typename _DataType>
...
     custom_blas_gemm_c(void* array1, void* array2, void* result1, size_t size_m, size_t size_n, size_t size_k);
 
+
+/**
+ * @ingroup BACKEND_API
+ * @brief absolute function.
+ *
+ * @param [in]  array1_in    Input array.
+ *
+ * @param [in]  input_shape  Input shape.
```
Comment (Alexander-Makaryev) [inline dpnp/backend/backend_iface.hpp:107]: @Rubtsowa Unnecessary param

File: dpnp/backend/backend_iface.hpp [hunk 6]
```diff
@@ -97,6 +97,26 @@ template <typename _DataType>
...
+ *
+ * @param [out] result1      Output array.
+ *
+ * @param [in]  size         Number of elements in input arrays.
+ *
+ */
+template <typename _DataType>
+INP_DLLEXPORT void custom_elemwise_absolute_c(void* array1_in,
+                                               const std::vector<long>& input_shape,
```
Comment (Alexander-Makaryev) [inline dpnp/backend/backend_iface.hpp:116]: @Rubtsowa Unnecessary param

File: dpnp/backend/custom_kernels_mathematical.cpp [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+                                                 void* result1,
+                                                 size_t size);
+template void custom_elemwise_absolute_c<long>(void* array1_in,
+                                                const std::vector<long>& input_shape,
+                                                void* result1,
+                                                size_t size);
+template void custom_elemwise_absolute_c<int>(void* array1_in,
+                                               const std::vector<long>& input_shape,
+                                               void* result1,
+                                               size_t size);
```
Comment (Alexander-Makaryev) [inline dpnp/backend/custom_kernels_mathematical.cpp:97]: @Rubtsowa input_shape is unnecessary param

File: dpnp/backend_mathematical.pyx [hunk 8]
```diff
@@ -63,54 +63,29 @@ __all__ += [
 ]
 
 
+ctypedef void(*fptr_custom_elemwise_absolute_1in_1out_t)(void *, dparray_shape_type &,
```
Comment (Alexander-Makaryev) [inline dpnp/backend_mathematical.pyx:66]: @Rubtsowa Unnecessary param

File: dpnp/backend_mathematical.pyx [hunk 9]
```diff
@@ -63,54 +63,29 @@ __all__ += [
...
+    # get the FPTR data structure
+    cdef DPNPFuncData kernel_data = get_dpnp_function_ptr(DPNP_FN_ABSOLUTE, param1_type, param1_type)
+
+    result_type = dpnp_DPNPFuncType_to_dtype( < size_t > kernel_data.return_type)
+    # ceate result array with type given by FPTR data
+    cdef dparray result = dparray(input_shape, dtype=result_type)
+
+    cdef fptr_custom_elemwise_absolute_1in_1out_t func = <fptr_custom_elemwise_absolute_1in_1out_t > kernel_data.ptr
+    # call FPTR function
+    func(input.get_data(), input_shape, result.get_data(), input.size)
```
Comment (Alexander-Makaryev) [inline dpnp/backend_mathematical.pyx:86]: @Rubtsowa Unnecessary param

File: dpnp/backend_mathematical.pyx [hunk 10]
```diff
@@ -63,54 +63,29 @@ __all__ += [
...
-            result_offset += (output_shape_offsets[i] * result_indexes_val)
-
-        input_elem = input.item(source_idx)
-        result_array[result_offset] = input_elem if input_elem >= 0 else -1 * input_elem
-
-    dpnp_array = dpnp.array(result_array, dtype=input.dtype)
-    dpnp_result_array = dpnp_array.reshape(output_shape)
-    return dpnp_result_array
+    cdef dparray_shape_type input_shape = input.shape
+    cdef size_t input_shape_size = input.ndim
```
Comment (Alexander-Makaryev) [inline dpnp/backend_mathematical.pyx:72]: @Rubtsowa input_shape_size is not used

File: dpnp/backend_mathematical.pyx [hunk 11]
```diff
@@ -63,54 +63,29 @@ __all__ += [
...
-        for i, result_indexes_val in enumerate(result_indexes):
-            result_offset += (output_shape_offsets[i] * result_indexes_val)
-
-        input_elem = input.item(source_idx)
-        result_array[result_offset] = input_elem if input_elem >= 0 else -1 * input_elem
-
-    dpnp_array = dpnp.array(result_array, dtype=input.dtype)
-    dpnp_result_array = dpnp_array.reshape(output_shape)
-    return dpnp_result_array
+    cdef dparray_shape_type input_shape = input.shape
```
Comment (Alexander-Makaryev) [inline dpnp/backend_mathematical.pyx:71]: @Rubtsowa Unnecessary param

**Pull request merged**